File: app/src/main/kotlin/com/example/unittesting/BasePresenter.java
package com.example.unittesting;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public class BasePresenter<T> implements Presenter<T> {

    T view;
    CompositeDisposable compositeDisposable = new CompositeDisposable();

    @Override
    public void createView(T view) {
        this.view = view;
    }

    @Override
    public void destroyView() {
        compositeDisposable.clear();
        view = null;
    }

    protected void bindToLifecycle(Disposable disposable) {
        compositeDisposable.add(disposable);
    }

    protected T getView() {
        return view;
    }
}

File: app/src/main/kotlin/com/example/unittesting/BasePresenter.kt
package com.example.unittesting

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

open class BasePresenter<T> : Presenter<T> {

    var view: T? = null
    private val compositeDisposable = CompositeDisposable()

    override fun createView(view: T) {
        this.view = view
    }

    fun bindToLifecycle(disposable: Disposable) {
        compositeDisposable.add(disposable)
    }

    override fun destroyView() {
        compositeDisposable.clear()
        view = null
    }
}
